### PR TITLE
Bump golang 1.18 and k8s 1.24

### PR DESCRIPTION
**Pull request by SchSeba**
Signed-off-by: Sebastian Sch <sebassch@gmail.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.13-alpine as builder
+FROM golang:alpine as builder
```
Comment (adrianchiris) [inline Dockerfile:15]: shall we stick to a specific golang version when building image (like 1.18.3-alpine)? 

https://hub.docker.com/_/golang

File: scripts/e2e_setup_cluster.sh [hunk 2]
```diff
@@ -153,7 +153,7 @@ retry kubectl -n kube-system wait --for=condition=ready -l name="${CNIS_NAME}" p
 echo "## install NRI"
 retry kubectl create -f "${root}/deployments/auth.yaml"
 # enable HugePage API at webhook
-awk '/-logtostderr/ { print; print "    - -injectHugepageDownApi"; next }1' "${root}/deployments/server.yaml" > "${root}/deployments/server_huge.yaml"
+awk '/-logtostderr/ { print; print "        - -injectHugepageDownApi"; next }1' "${root}/deployments/server.yaml" > "${root}/deployments/server_huge.yaml"
```
Comment (adrianchiris) [inline scripts/e2e_setup_cluster.sh:156]: nit: accidental spaces ?

Comment (SchSeba) [reply scripts/e2e_setup_cluster.sh:156]: nope the lane was failing because of bad indentation so I fix it

File: Dockerfile (hunk 1 above)
Comment (SchSeba) [reply Dockerfile:15]: works for me :)

**Review by adrianchiris — APPROVED** [review]
Thanks for addressing the comments !

**Pull request merged**